test/reflect.js

```diff
@@ -0,0 +1,483 @@
+/*global describe, it, expect, require, Reflect */
+
+var exported = require('../');
+
+describe('Reflect', function () {
+  var object = {
+    something: 1,
+    _value: 0
+  };
+
+  Object.defineProperties(object, {
+    value: {
+      get: function () {
+        return this._value;
+      }
+    },
+
+    setter: {
+      set: function (val) {
+        this._value = val;
+      }
+    },
+
+    bool: {
+      value: true
+    }
+  });
+
+  var testCallableThrow = function (func) {
+    [null, undefined, 1, 'string', true, [], {}].forEach(function (item) {
+      expect(func.bind(null, item)).to.throw(TypeError);
```

Comment:
`function () { func(item); }` would be a bit faster here, and wouldn't look significantly messier imo. Just a thought.

Either way, `func.bind(null, item)` or `function () { func(item); }` should be defined outside of the loop, to avoid creating a function in each iteration. Same below.